File: OrderViewer/ClientApp/app/models/list.response.ts
export interface IListResponse<T> {
    model: T[];
    message: string;
    didError: boolean;
    errorMessage: string;
}

export class ListResponse<T> implements IListResponse<T> {
    public model: T[];
    public message: string;
    public didError: boolean;
    public errorMessage: string;
}

File: Legacy/OrderViewer/Responses/ListResponse.cs
﻿using System;
using System.Collections.Generic;

namespace OrderViewer.Responses
{
    public class ListResponse<TModel> : IListResponse<TModel>
    {
        public String Message { get; set; }

        public Boolean DidError { get; set; }

        public String ErrorMessage { get; set; }

        public IEnumerable<TModel> Model { get; set; }

        public Int32 PageSize { get; set; }

        public Int32 PageNumber { get; set; }

        public Int32 ItemsCount { get; set; }

        public Int32 PageCount =>
            PageSize == 0 ? 0 : ItemsCount / PageSize;
    }
}
